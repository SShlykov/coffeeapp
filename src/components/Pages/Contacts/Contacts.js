import React, {Component} from "react";
import cofmaf from "./img/CofMaf.png";
import arrowLeft from "./img/ArrowLeft.png"
import {BeansLogo} from "../../SmallComp";

export default class Contacts extends Component {
    state ={
        form: true
    };
    submitFunc = (e) =>{
      e.preventDefault();
      this.setState({
          form: false
      })
    };
    anotherMes = () => {
      this.setState({
          form: true
      })
    };
    render() {
        let form = () => {
            return (
                <form  onSubmit={this.submitFunc} className="contact-form">
                    <label htmlFor="name" className="col-6 contact-label importantInput">Name</label>
                    <input type="name" id="name" className="col-6 contact-input"/>
                    <label htmlFor="e-mail" className="col-6 contact-label importantInput">E-mail</label>
                    <input type="e-mail" id="e-mail" className="col-6 contact-input"/>
                    <label htmlFor="phone" className="col-6 contact-label">Phone</label>
                    <input type="phone" id="phone" placeholder="+7(...)...-...." className="col-6 contact-input"/>
                    <label htmlFor="message" id="labelMes" className="col-12 text-center contact-label importantInput">Your message</label>
                    <textarea placeholder="Tell us..." name="message" id="message" rows="4" className="col-12"/>
                    <div className="centred">
                        <input type="submit" value="Send us" className="anotherButton"/>
                    </div>
                </form>
            )
        };
        let thanks = () => {
            return(
                <div className="thanksBlock">
                    <h2 className="title thanksTitle">
                        Thank you so much<br/>
                        We contact you as soon as posible
                    </h2>
                    <div className="cofmafImage">
                        <img src={cofmaf} alt="icon"/>
                    </div>
                    <div className="centred">
                        <button className="anotherButton" onClick={this.anotherMes}>Another ?&nbsp; <img src={arrowLeft} alt="â†©"/></button>
                    </div>
                </div>
            )
        };
        let content = (this.state.form)? form(): thanks();
        return(
            <>
                <div className="container">
                    <div className="row">
                        <div className="col-lg-6 offset-lg-3">
                            <div className="contacts-title">
                                <div className="title bold">Tell us about your tastes</div>
                                <BeansLogo dark/>
                            </div>
                            {content}
                        </div>
                    </div>
                </div>
            </>
        )
    }

}