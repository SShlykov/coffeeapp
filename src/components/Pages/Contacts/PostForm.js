import React, {Component} from "react";
import cDataService from "../../../services/cDataService";

export default class Form extends Component {
    state = {
      name: "",
      email: "",
      phone: "",
      message: "",
      data: {},
    };
    cDataService = new cDataService();
    onType = ( e ) => {
        let val = e.target.value;
        let attr = e.target.id;
        this.setState({[attr]:val});
    };
    updateDataObj = (name, email, phone, message)=> {
        this.setState({
            data: {name:name, email:email, phone:phone, message: message}
        });
    };
    submitFunc = async (e) =>{
        const {name, email, phone, message} = this.state;
        e.preventDefault();
        await this.updateDataObj(name, email, phone, message);
        await this.cDataService.postMessage(this.state.data);
        await this.props.submitListener();
    };
    render() {
        const  {name, message, phone, email} = this.state;
        return (
            <form onSubmit={this.submitFunc} method="POST" className="contact-form">
                <label htmlFor="name" className="col-6 contact-label importantInput">Name</label>
                <input
                    type="name"
                    id="name"
                    className="col-6 contact-input"
                    value={name}
                    onChange={this.onType}
                />
                <label htmlFor="e-mail" className="col-6 contact-label importantInput">E-mail</label>
                <input
                    type="e-mail"
                    id="email"
                    className="col-6 contact-input"
                    value={email}
                    onChange={this.onType}
                />
                <label htmlFor="phone" className="col-6 contact-label">Phone</label>
                <input
                    type="phone"
                    id="phone"
                    placeholder="+7(...)...-...."
                    className="col-6 contact-input"
                    value={phone}
                    onChange={this.onType}
                />
                <label htmlFor="message" id="labelMes" className="col-12 text-center contact-label importantInput">Your
                    message</label>
                <textarea
                    placeholder="Tell us..."
                    name="message"
                    id="message"
                    className="col-12"
                    value={message}
                    onChange={this.onType}
                />
                <div className="centred">
                    <input
                        type="submit"
                        value="Send us"
                        className="anotherButton"
                    />
                </div>
            </form>
        )
    }
};
