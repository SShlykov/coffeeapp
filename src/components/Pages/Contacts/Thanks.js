import React from "react";
import cofmaf from "./img/CofMaf.png";
import arrowLeft from "./img/ArrowLeft.png";

let Thanks = ({anotherMes}) => {
    return(
        <div className="thanksBlock">
            <h2 className="title thanksTitle">
                Thank you so much<br/>
                We contact you as soon as posible
            </h2>
            <div className="cofmafImage">
                <img src={cofmaf} alt="icon"/>
            </div>
            <div className="centred">
                <button className="anotherButton" onClick={() => anotherMes()}>Another ?&nbsp; <img src={arrowLeft} alt="â†©"/></button>
            </div>
        </div>
    )
};
export default Thanks;