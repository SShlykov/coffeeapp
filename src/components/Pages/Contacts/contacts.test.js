import React from 'react';
import Contacts from "./contacts";
import {shallow} from 'enzyme';

describe("Testing <Contacts />", () => {
    const page = shallow(<Contacts/>);
    describe("Testing snap & state", ()=>{
        it("Contacts has rendered correctly", () => {
            expect(page).toMatchSnapshot();
        });
        it("RandomChar state 'form' is true" , ()=>{
            expect(page.state().form).toBeTruthy();
        })
    })
    describe("Handlers tests", () => {
        it("testing submitListener", () => {
            page.instance().submitListener();
            expect(page.state().form).toBeFalsy();
        })
        it("testing anotherMes", () => {
            page.instance().submitListener();
            page.instance().anotherMes();
            expect(page.state().form).toBeTruthy();
        })
    })
})