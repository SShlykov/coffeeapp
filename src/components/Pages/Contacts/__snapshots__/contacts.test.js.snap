// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Contacts /> Testing snap & state Contacts has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contacts />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-6 offset-lg-3"
          >
            <div
              className="contacts-title"
            >
              <div
                className="title bold"
              >
                Tell us about your tastes
              </div>
              <BeansLogo
                dark={true}
              />
            </div>
            <Form
              submitListener={[Function]}
            />
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-lg-6 offset-lg-3"
          >
            <div
              className="contacts-title"
            >
              <div
                className="title bold"
              >
                Tell us about your tastes
              </div>
              <BeansLogo
                dark={true}
              />
            </div>
            <Form
              submitListener={[Function]}
            />
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-lg-6 offset-lg-3"
          >
            <div
              className="contacts-title"
            >
              <div
                className="title bold"
              >
                Tell us about your tastes
              </div>
              <BeansLogo
                dark={true}
              />
            </div>
            <Form
              submitListener={[Function]}
            />
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="contacts-title"
              >
                <div
                  className="title bold"
                >
                  Tell us about your tastes
                </div>
                <BeansLogo
                  dark={true}
                />
              </div>,
              <Form
                submitListener={[Function]}
              />,
            ],
            "className": "col-lg-6 offset-lg-3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="title bold"
                  >
                    Tell us about your tastes
                  </div>,
                  <BeansLogo
                    dark={true}
                  />,
                ],
                "className": "contacts-title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tell us about your tastes",
                    "className": "title bold",
                  },
                  "ref": null,
                  "rendered": "Tell us about your tastes",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "dark": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "submitListener": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-6 offset-lg-3"
            >
              <div
                className="contacts-title"
              >
                <div
                  className="title bold"
                >
                  Tell us about your tastes
                </div>
                <BeansLogo
                  dark={true}
                />
              </div>
              <Form
                submitListener={[Function]}
              />
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-lg-6 offset-lg-3"
            >
              <div
                className="contacts-title"
              >
                <div
                  className="title bold"
                >
                  Tell us about your tastes
                </div>
                <BeansLogo
                  dark={true}
                />
              </div>
              <Form
                submitListener={[Function]}
              />
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-lg-6 offset-lg-3"
            >
              <div
                className="contacts-title"
              >
                <div
                  className="title bold"
                >
                  Tell us about your tastes
                </div>
                <BeansLogo
                  dark={true}
                />
              </div>
              <Form
                submitListener={[Function]}
              />
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="contacts-title"
                >
                  <div
                    className="title bold"
                  >
                    Tell us about your tastes
                  </div>
                  <BeansLogo
                    dark={true}
                  />
                </div>,
                <Form
                  submitListener={[Function]}
                />,
              ],
              "className": "col-lg-6 offset-lg-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="title bold"
                    >
                      Tell us about your tastes
                    </div>,
                    <BeansLogo
                      dark={true}
                    />,
                  ],
                  "className": "contacts-title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Tell us about your tastes",
                      "className": "title bold",
                    },
                    "ref": null,
                    "rendered": "Tell us about your tastes",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "dark": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "submitListener": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
